### ISSUE TITLE
Have a command that clears all problems that are from tasks.

### ISSUE THREAD

**Issue by alexr00** [microsoft/vscode#77214]
When a user makes a change to a task and re-runs it, they could end up accumulating problems since the modified task counts as a new task.

One idea is to  clear all problems when the task config changes: https://github.com/microsoft/vscode/issues/76108
But that seems too aggressive and we'd be removing problems without telling the user why.

Instead, having a command to clear the task problems is clearer.

**Comment by egamma** [comment]
@alexr00 pls see also the discussion here #50448

**Comment by justingrant** [comment]
@alexr00 - it would be great if this command could accept an optional task name parameter, so that users could clear problems from one specific task without clearing problems from all tasks.  In complex projects with many different tasks that contribute problems, it's painful to have to re-run every task just to clear out bad problems from one task.  Another reason for this is when you quit a background task, there's no way to clear its problems.

**Comment by alexr00** [comment]
When you quick a background task it's errors should be cleaned up.

Passing some parameter to specify which problems to clear seems reasonable.

**Comment by alexr00** [comment]
@sandy081 there is not a great way to clear only markers from tasks right now since a marker from a task can have the same owner as a marker from someone else. Since the marker `remove` method takes an `owner` and a `resource` this isn't possible. 

What do you think of having a sub-owner for markers or some other property that can be passed into `remove` to filter which markers are removed?

**Comment by sandy081** [comment]
Are you going through the public or internal API?

**Comment by alexr00** [comment]
@sandy081 internal API.

**Comment by sandy081** [comment]
May be we can change current `owner` to `source` and introduce new `owner` property.

@jrieken Let me know if you have other suggestions?

**Comment by jrieken** [comment]
This isn't sooo easy because markers are double-indexed by owner and by resource. So, this will definitely a slow and error-prune approach. @alexr00 Why is it that tasks re-use owners?

**Comment by alexr00** [comment]
@jrieken it is done so that you don't get duplicate errors from a language server. For example, the typescript language server shows you errors for files you have open. If you have a build task it will create markers for all errors, including the ones that the typescript language server already uses. To prevent this, we use the same owner.

**Comment by jrieken** [comment]
Yeah, that makes sense. It just won't be super easy to implement because of the way the marker service works today.

**Comment by NSExceptional** [comment]
Really hope this gets picked up, +1

**Comment by DavidBiesack** [comment]
Rather than stale problems from a task that I've run, stale problems show up most often for me with the [Spectral](https://github.com/stoplightio/vscode-spectral) or yamllint extensions that do not always clean up problems, such as the problem
```
[{
	"resource": ".../path.../openapi.yaml",
	"owner": "spectral",
	"code": "invalid-ref",
	"severity": 8,
	"message": "'#/components/responses/401' does not exist",
	"source": "spectral",
	"startLineNumber": 684,
	"startColumn": 17,
	"endLineNumber": 684,
	"endColumn": 45
}]
```
I fix the problems, but the extension does not properly remove the old items. I've [reported this via the extension](https://github.com/stoplightio/vscode-spectral/issues/171).

But until the extensions are fixed, 
(and I've seen this problem with other extensions) I'd like to be able to clear these problems from the view.
without having to restart VS Code or close/reopen the VS Code window

**Comment by starball5** [comment]
Related on Stack Overflow: [VS Code Problems tab: Clear items coming from tasks with problemMatcher](https://stackoverflow.com/q/77750818/11107541).

Somewhat related: #66982, #73186.

**Comment by heartacker** [comment]
weneedthisindeed
